plugin/trino-mariadb/src/test/java/io/trino/plugin/mariadb/BaseMariaDbTableStatisticsTest.java

```diff
@@ -0,0 +1,443 @@
+        executeInMariaDb("CREATE OR REPLACE VIEW " + tableName + " AS SELECT orderkey, custkey, orderpriority, comment FROM orders");
+        try {
+            assertQuery(
+                    "SHOW STATS FOR " + tableName,
+                    "VALUES " +
+                            "('orderkey', null, null, null, null, null, null)," +
+                            "('custkey', null, null, null, null, null, null)," +
+                            "('orderpriority', null, null, null, null, null, null)," +
+                            "('comment', null, null, null, null, null, null)," +
+                            "(null, null, null, null, null, null, null)");
+            // It's not possible to ANALYZE a VIEW in MariaDB
+        }
+        finally {
+            executeInMariaDb("DROP VIEW " + tableName);
+        }
+    }
+
+    @Test
+    @Override
+    public void testMaterializedView()
+    {
+        throw new SkipException(""); // TODO is there a concept like materialized view in MariaDB?
+    }
+
+    @Test(dataProvider = "testCaseColumnNamesDataProvider")
+    @Override
+    public void testCaseColumnNames(String tableName)
+    {
+        executeInMariaDb(("" +
+                "CREATE TABLE " + tableName + " " +
+                "AS SELECT " +
+                "  orderkey AS CASE_UNQUOTED_UPPER, " +
+                "  custkey AS case_unquoted_lower, " +
+                "  orderstatus AS cASe_uNQuoTeD_miXED, " +
+                "  totalprice AS \"CASE_QUOTED_UPPER\", " +
+                "  orderdate AS \"case_quoted_lower\"," +
+                "  orderpriority AS \"CasE_QuoTeD_miXED\" " +
+                "FROM orders")
+                .replace("\"", "`"));
+        try {
+            gatherStats(tableName);
+            MaterializedResult statsResult = computeActual("SHOW STATS FOR " + tableName);
+            assertColumnStats(statsResult, new MapBuilder<String, Integer>()
+                    .put("case_unquoted_upper", 15000)
+                    .put("case_unquoted_lower", 1000)
+                    .put("case_unquoted_mixed", varcharNdvToExpected.apply(3))
+                    .put("case_quoted_upper", 14996)
+                    .put("case_quoted_lower", 2401)
+                    .put("case_quoted_mixed", varcharNdvToExpected.apply(5))
+                    .build());
+            assertThat(getTableCardinalityFromStats(statsResult)).isCloseTo(15000, withinPercentage(20));
+        }
+        finally {
+            executeInMariaDb("DROP TABLE " + tableName.replace("\"", "`"));
+        }
+    }
+
+    @Test
+    @Override
+    public void testNumericCornerCases()
+    {
+        try (TestTable table = fromColumns(
+                getQueryRunner()::execute,
+                "test_numeric_corner_cases_",
+                ImmutableMap.<String, List<String>>builder()
+                        // TODO Infinity and NaNs not supported by MySQL. Are they not supported in MariaDB as well?
```

Comment:
I'd look at type mapping tests to find the answer.

Reply:
good point.

@homar please follow-up here. See all `// TODO Infinity and NaNs not supported by MySQL` occurrences in the code.